### PR TITLE
feat(ts): convert aboutmodal to TS

### PR DESCRIPTION
**Pull request by redallen**
**What**:

**Additional issues**:

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/AboutModal/index.js [hunk 1]
```diff
@@ -1 +0,0 @@
-export { default as AboutModal } from './AboutModal';
```
Comment (seanforyou23) [inline packages/patternfly-4/react-core/src/components/AboutModal/index.js]: Please rename these files with git instead of deleting and re-adding so we can better preserve the history.

AboutModal.js => AboutModal.tsx
AboutModal.test.js => AboutModal.test.tsx
AboutModalBox.js => AboutModalBox.tsx
AboutModalBoxBrand.js => AboutModalBoxBrand.tsx
AboutModalBoxCloseButton.js => AboutModalBoxCloseButton.tsx
AboutModalBoxContent.js => AboutModalBoxContent.tsx
AboutModalBoxHeader.js => AboutModalBoxHeader.tsx
AboutModalBoxHero.js => AboutModalBoxHero.tsx
AboutModalContainer.js => AboutModalContainer.tsx
index.js => index.ts

File: packages/patternfly-4/react-core/scripts/convertComponentHelpers.js [hunk 2]
```diff
@@ -1,98 +1,87 @@
 function getMatches(string, pattern) {
```
Comment (seanforyou23) [inline packages/patternfly-4/react-core/scripts/convertComponentHelpers.js]: Seems like the changes to convertComponentHelpers.js and convertComponentToTS.js would be better served as part of an independent PR.

File: packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    trademark: '',
+    backgroundImageSrc: '',
+    noAboutModalBoxContentContainer: false
+  };
+
+  constructor(props: AboutModalProps) {
+    super(props);
+    this.state = {
+      container: undefined,
+      ariaLabelledBy: `pf-about-modal-title-${AboutModal.currentId++}`,
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx]: Can you remove these from the state.  They are not stateful.

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBox.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  /** additional classes added to the AboutModalBox  */
+  className?: string;
+};
+
+export const AboutModalBox: React.SFC<AboutModalBoxProps> = ({
+  children,
+  className = '',
+  ...props
+}: AboutModalBoxProps) => (
+  <div {...props} role="dialog" aria-modal="true" className={css(styles.aboutModalBox, className)}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBox.tsx]: It probably does not make a difference since it is internal, but can you spread the props at the end here since you removed the aria attributes from the props so the overwrite any other props.

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBoxCloseButton.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  /** A callback for when the close button is clicked  */
+  onClose?: () => void;
+};
+
+export const AboutModalBoxCloseButton: React.SFC<AboutModalBoxCloseButtonProps> = ({
+  className = '',
+  onClose = () => undefined,
+  ...props
+}: AboutModalBoxCloseButtonProps) => (
+  <div {...props} className={css(styles.aboutModalBoxClose, className)}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBoxCloseButton.tsx]: can you spread props at the end please.

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBoxContent.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+export const AboutModalBoxContent: React.SFC<AboutModalBoxContentProps> = ({
+  children,
+  className = '',
+  trademark,
+  id,
+  noAboutModalBoxContentContainer = false,
+  ...props
+}: AboutModalBoxContentProps) => (
+  <div {...props} className={css(styles.aboutModalBoxContent, className)} id={id}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalBoxContent.tsx]: Can you spread props at the end please

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  /** additional classes added to the About Modal Box  */
+  className?: string;
+  /** Flag to show the About Modal  */
+  isOpen?: boolean;
+  /** A callback for when the close button is clicked  */
+  onClose?: () => void;
+  /** Product Name  */
+  productName?: string;
+  /** Trademark information  */
+  trademark?: string;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx]: Is there a reason this is no longer required?

File: packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/patternfly/components/Backdrop/backdrop.css';
+import { canUseDOM } from 'exenv';
+import { KEY_CODES } from '../../helpers/constants';
+import { AboutModalContainer } from './helpers/AboutModalContainer';
```
[the remark this answers is not in the document]
Comment (redallen) [reply packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx]: I'll move the files up a directory to be like all the other components.

File: packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx (hunk 3 above)
Comment (redallen) [reply packages/patternfly-4/react-core/src/components/AboutModal/AboutModal.tsx]: I'll use class attributes instead.

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx (hunk 7 above)
Comment (redallen) [reply packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx]: Yes. I removed it to be consistent with `AboutModal` (the component consumers use), which does not require it. Otherwise a type error gets thrown in `AboutModal.tsx` when it constructs an `AboutModalContainer`.

File: packages/patternfly-4/react-core/scripts/convertComponentHelpers.js (hunk 2 above)
Comment (redallen) [reply packages/patternfly-4/react-core/scripts/convertComponentHelpers.js]: Reverted.

File: packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx (hunk 7 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/AboutModal/helpers/AboutModalContainer.tsx]: Nice catch.
